File: JavaScript/BOJ_Solving/Bronze/2588번： 곱셈/곱셈.js
"use strict";

const fs = require("fs");
let input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map(Number);

let one = input[0];
let two = input[1];
let three;
let four;
let five;

let twoArr = two.toString();

for (let i = 0; i < twoArr.length; i++) {
  if (i === 2) {
    three = one * twoArr[i];
  }
  if (i === 1) {
    four = one * twoArr[i];
  }
  if (i === 0) {
    five = one * twoArr[i];
  }
}

let result = one * two;

let arr = [three, four, five, result];

for (let i = 0; i < arr.length; i++) {
  console.log(arr[i]);
}

File: Python/BOJ_Solving/Silver/1629번： 곱셈/곱셈.py
import sys

a, b, c = map(int, sys.stdin.readline().split())

def power_mod(a, b, c):
    if b == 0:
        return 1 
    
    if b == 1:
        return a % c
    
    x = power_mod(a, b // 2, c)
    x = (x * x) % c 
    
    if b % 2 == 0:
        return x
    else:
        return (x * a) % c

result = power_mod(a, b, c)
print(result)
